File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int  n = prices.length;
        int profit = 0;
        int low = prices[0];
        for(int i = 0;i<n;i++){
            low = Math.min(low,prices[i]);
            profit = Math.max(profit,prices[i] - low);
        }
        return profit;
    }
}

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        n = len(prices)
        profit = 0
        low = prices[0]
        for i in range(0,n):
            low = min(low,prices[i])
            profit = max(profit,prices[i] - low)
        return profit